src/utils/app-common.tsx

```diff
@@ -0,0 +1,10 @@
+// Functions used by both animations and simulations
+
+export const delayControl =
+  (setControl: (val: boolean) => void, setDisableControls: (val: boolean) => void, delay?: number) => (
+    (controlVal: boolean) => {
+      setDisableControls(true);
+      setControl(controlVal);
+      setTimeout(() => setDisableControls(false), delay || 750);
```

Comment:
`750` seems like a long time -- did you try with smaller values, e.g. `500`?

Reply:
I started with 500 and worked up. It's one of those things where the longer the better, but the system can enter a broken state at any duration. What would be best would be to figure out why the background video doesn't always work, but I don't think I can dedicate more time to it now. My guess is that sometimes the back video hasn't gotten updated by the time the front video changes again, and then it falls into some sort of limbo state.

For now, I dropped the duration to 600, and made the control animation take the same amount of time. I think this should give the user feedback that they can't change the switch again until the switch is finished, which I hope should mitigate some frustration.